### PR TITLE
add input option for long-range method in CCZ force estimator

### PR DESCRIPTION
**Pull request by Paul-St-Young**
Allow input such as:
`<estimator name="myforce" type="Force" mode="cep" lrmethod="srcoul">`
to use optimized breakup rather than Ewald breakup the long-range Coulomb contribution to forces.

@rcclay may I clean up `ForceChiesaPBCAA` a bit?
There is a lot of commented-out code. I am not sure if any of it is still useful.

### CODE REVIEW

**Comment by markdewing** [comment]
okay to test

**Comment by rcclay** [comment]
@Paul-St-Young chop away, and chop with vigor.

**Comment by prckent** [comment]
@Paul-St-Young  Wholesale removal of commented and legacy code is encouraged.

Put a comment here when you are done. Thanks.

**Comment by Paul-St-Young** [comment]
@rcclay @prckent I am done with the cleaning. All unit tests are green on my machine. This PR is good to go if all the checks pass.

File: manual/hamiltonianobservable.tex [hunk 1]
```diff
@@ -1385,6 +1385,7 @@ \section{``Force'' estimators} \label{sec:force_est}
 \multicolumn{2}{l}{attributes}  & \multicolumn{4}{l}{}\\
    & \bfseries name       & \bfseries datatype & \bfseries values  & \bfseries default   & \bfseries description \\
    &   \texttt{mode}$^o$      &  text              & \textit{See above}        & bare          & Select estimator type\\
+   &   \texttt{lrmethod}$^o$  &  text              & "ewald" or "srcoul"       & ewald         & Select long-range potential breakup method\\
```
Comment (ye-luo) [inline manual/hamiltonianobservable.tex]: Remove quotes

File: src/QMCHamiltonians/ForceChiesaPBCAA.h [hunk 2]
```diff
@@ -135,6 +114,7 @@ struct ForceChiesaPBCAA : public OperatorBase, public ForceBase
   // AA table ID
   const int d_aa_ID;
   const int d_ei_ID;
+  std::string lrmethod;
```
Comment (ye-luo) [inline src/QMCHamiltonians/ForceChiesaPBCAA.h]: Please add documentation.

**Comment by prckent** [comment]
Thanks Paul!

**Comment by prckent** [comment]
Final comment/aside: sometime in the future when we have a proper grip on forces, we should automate the Coulomb convergence settings. e.g. via an input convergence tolerance. This would reduce the risk of surprises and mistakes.

**Pull request merged**